nwb_conversion_tools/utils/spike_interface.py

```diff
@@ -1013,177 +944,86 @@ def write_units(
-                unit_kwargs.update({pr: prop_value})
-            else:  # Case of missing data for this unit and this property
-                unit_kwargs.update({pr: np.nan})
-        units_table.add_unit(id=int(unit_id), spike_times=spkt, **unit_kwargs)
-
-        # TODO
-        # # Stores average and std of spike traces
-        # This will soon be updated to the current NWB standard
-        # if 'waveforms' in sorting.get_unit_spike_feature_names(unit_id=id):
-        #     wf = sorting.get_unit_spike_features(unit_id=id,
-        #                                          feature_name='waveforms')
-        #     relevant_ch = most_relevant_ch(wf)
-        #     # Spike traces on the most relevant channel
-        #     traces = wf[:, relevant_ch, :]
-        #     traces_avg = np.mean(traces, axis=0)
-        #     traces_std = np.std(traces, axis=0)
-        #     nwbfile.add_unit(
-        #         id=id,
-        #         spike_times=spkt,
-        #         waveform_mean=traces_avg,
-        #         waveform_sd=traces_std
-        #     )
-
-    # Check that multidimensional features have the same shape across units
-    feature_shapes = dict()
-    for ft in all_features:
-        shapes = []
-        for unit_id in unit_ids:
-            if ft in sorting.get_unit_spike_feature_names(unit_id):
-                feat_value = sorting.get_unit_spike_features(unit_id, ft)
-                if isinstance(feat_value[0], (int, np.integer, float, str, bool)):
-                    break
-                elif isinstance(feat_value[0], (list, np.ndarray)):  # multidimensional features
-                    if np.array(feat_value).ndim > 1:
-                        shapes.append(np.array(feat_value).shape)
-                        feature_shapes[ft] = shapes
-                elif isinstance(feat_value[0], dict):
-                    print(f"Skipping feature '{ft}' because dictionaries are not supported.")
-                    skip_features.append(ft)
-                    break
+            if prop in ["max_channel", "max_electrode"]:
+                if nwbfile.electrodes is None:
+                    warnings.warn(
+                        "first link a RX to the nwb file to create correct electrodes")
+                    continue
+                assert set(data).issubset(
+                    set(nwbfile.electrodes.id.data)
+                ), "sorting and recording extractor should be for the same data"
+                unit_columns[prop].update(table=nwbfile.electrodes)
+
+    # 2. fill with provided custom descriptions
+    for x in metadata["units"]:
+        if x["name"] not in list(unit_columns):
+            raise ValueError(
+                f'"{x["name"]}" not a property of sorting object, set it first and rerun')
+        unit_columns[x["name"]]["description"] = x["description"]
+
+    # 3. For existing electrodes table, add the additional columns and fill with default data:
+    add_properties_to_dynamictable(
```

Comment:
We need to change this function to allow to add property to a dynamic table created outside of an NWBFile